zboxcore/sdk/writemarker_mutex.go

```diff
@@ -60,257 +52,215 @@ func CreateWriteMarkerMutex(client *client.Client, allocationObj *Allocation) (*
 
 	return &WriteMarkerMutex{
 		allocationObj: allocationObj,
-		zbox:          sdks.New(client.ClientID, client.ClientKey, client.SignatureScheme, client.Wallet),
 	}, nil
 }
 
-// Lock acquire WriteMarker lock from blobbers
-func (m *WriteMarkerMutex) Lock(ctx context.Context, connectionID string) error {
-	if m == nil {
-		return errors.Throw(constants.ErrInvalidParameter, "WriteMarkerMutex")
-	}
-	m.mutex.Lock()
-	defer m.mutex.Unlock()
-
-	if m.allocationObj == nil {
-		return errors.Throw(constants.ErrInvalidParameter, "allocationObj")
-	}
+func (wmMu *WriteMarkerMutex) Unlock(
+	ctx context.Context, mask zboxutil.Uint128,
+	blobbers []*blockchain.StorageNode,
+	timeOut time.Duration, connID string,
+) {
+	wg := &sync.WaitGroup{}
+	var pos uint64
+	for i := mask; !i.Equals64(0); i = i.And(zboxutil.NewUint128(1).Lsh(pos).Not()) {
+		pos = uint64(i.TrailingZeros())
 
-	T := len(m.allocationObj.Blobbers)
+		blobber := blobbers[pos]
 
-	if T == 0 {
-		return errors.Throw(constants.ErrInvalidParameter, "blobbers")
+		wg.Add(1)
+		go wmMu.UnlockBlobber(ctx, blobber, connID, timeOut, wg)
 	}
 
-	urls := make([]string, T)
-
-	builder := &strings.Builder{}
-	for i, b := range m.allocationObj.Blobbers {
-		builder.Reset()
-		builder.WriteString(strings.TrimRight(b.Baseurl, "/")) //nolint: errcheck
-		builder.WriteString(blobber.EndpointWriteMarkerLock)
-		builder.WriteString(m.allocationObj.Tx)
-		urls[i] = builder.String()
-	}
-
-	//protocol detail is on https://github.com/0chain/blobber/wiki/Features-Upload#upload
-	M := int(math.Ceil(float64(T) / float64(3) * float64(2))) //the minimum of M blobbers must accpet the marker
+	wg.Wait()
+}
 
-	//retry 3 times
-	for retry := 0; ; retry++ {
-		i := 1
-		n := 0
-		m.lockedBlobbers = nil
+// Change status code to 204
+func (wmMu *WriteMarkerMutex) UnlockBlobber(
+	ctx context.Context, b *blockchain.StorageNode,
+	connID string, timeOut time.Duration, wg *sync.WaitGroup,
+) {
```

Comment:
We should not ignore the error and just log it. Any one failed below should be noted by the caller.